main.cpp

```diff
@@ -197,261 +308,351 @@ struct RolandTB303
         float randomisationAmount = 4.f;
 
         void savingSequence(int stepSize, int numberOfStepsEnabled, int numberOfLoops, int tempo);
-
-        bool savingToExternal(int stepCount, std::string fileType = "TB303.p"); // NOTE: function checks whether external device is selected for export.
-
+        bool savingToExternal(int stepCount, std::string fileType = "TB303.p");
         void displaySavingProgress(int timeRemaining, float savingRate);
     };
 
     SavePattern save;
 
-// 3 things it can do:
-//     1) program sequence of notes.
     void programSequence(RolandTB303 sequence);
-
-//     2) apply filtering.
     void filterSignal(float cutoffFrequency);
-
-//     3) increase || decrease playback tempo.
-    float tempoAdjust(RolandTB303 tempo); // NOTE: returns value relating to amount of temp adjustment.
+    float tempoAdjust(RolandTB303 tempo);
 };
 
+void RolandTB303::SavePattern::savingSequence(int stepSize, int numberOfStepsEnabled, int numberOfLoops, int tempo)
+{
+    int savedSeq = stepSize + numberOfStepsEnabled + numberOfLoops + tempo;
+    std::cout << "Saving sequence.../n" << savedSeq;
+}
+
+bool RolandTB303::SavePattern::savingToExternal(int stepCount, std::string fileType)
+{
+    if(stepCount > 0)
+    {
+        std::cout << "Exporting " << fileType;
+        return true;
+    }
+    else
+    {
+        std::cout << "Export error, no steps";
+        return false;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise

Reply:
Thank you! That's given me a good tool to approach these statements with.